spectator-ext-aws/src/main/java/com/netflix/spectator/aws/SpectatorRequestMetricCollector.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2015 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spectator.aws;
+
+import com.amazonaws.Request;
+import com.amazonaws.Response;
+import com.amazonaws.metrics.RequestMetricCollector;
+import com.amazonaws.util.AWSRequestMetrics;
+import com.amazonaws.util.TimingInfo;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spectator.api.Timer;
+import com.netflix.spectator.impl.Preconditions;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A RequestMetricCollector that captures request level metrics in a Registry.
+ */
+public class SpectatorRequestMetricCollector extends RequestMetricCollector {
+    private final Registry registry;
+
+    /**
+     * Constructs a new SpectatorRequestMetricCollector.
+     *
+     * @param registry the Registry for this SpectatorRequestMetricCollector.
+     */
+    public SpectatorRequestMetricCollector(Registry registry) {
+        super();
+        this.registry = Preconditions.checkNotNull(registry, "registry");
+    }
+
+    @Override
+    public void collectMetrics(Request<?> request, Response<?> response) {
+        final AWSRequestMetrics metrics = request.getAWSRequestMetrics();
+        if (metrics.isEnabled()) {
+            final String[] tags = getTags(request, response);
+            metrics.getTimingInfo().getAllCounters().forEach((name, count) -> {
+                registry.gauge(registry.createId(counterName(name), tags), count);
```

Comment:
The counter values do seem to be more long-lived stats around the underlying Apache HttpClient resources, and the TimingInfo counter API supports both a set and increment so definitely not monotonically increasing.

It might actually be a combination of values that would be gauge-like vs counter-like.

I was trying to avoid adding any knowledge or expectation about the names in here so it wouldn't have to care about or be updated when or if the SDK changed, but there is a MetricType enum (AWSRequestMetrics.Field) - I could potentially just pick the set of items worth tracking out of there and then explicitly pick those out in here instead of scraping the whole counter and timing metric collections